src/Workspaces/Core/Portable/Remote/RemoteUtilities.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    using DocumentTextChanges = ImmutableArray<(DocumentId, ImmutableArray<TextChange>)>;
+
+    internal static class RemoteUtilities
+    {
+        public static async Task<DocumentTextChanges> GetDocumentTextChangesAsync(
```

Comment:
because normally we can just pass around solutions when in the same process.  But for remote/oop we want to be able to say "here's the solution result of this operation" and we do that by passing along hte text changes necessary to convert the original solutoin to the same one that the OOP side see.s